### DISCUSSION TITLE
Potential Pi Pico MacOS enumeration issue

### DISCUSSION THREAD

**Discussion by savt22** [hathach/tinyusb#1229] [Q&A]
Hi, 

I hope this is the right place. I have made a MIDI controller using the Pimoroni Pico RGB Keypad base (https://github.com/savt22/2040_midi_chord_player). It uses CircuitPython 7, and the Adafruit_midi library.

It works as expected on Windows, and if I flash the Pico and code the Pico using sudo nano code.py on MacOS, it works as anticipated.

However, as soon as I do a hard reset or restart my Mac, the Pico is no longer recognised. I then can't get the Mac to recognise it again.

On the Console, 6 kernel errors appear when plugging the programmed Pico:

130345.648474 IOUSBHostDevice@02241000: IOUSBHostDevice::start_block_invoke: device descriptor fragment is invalid

130345.651657 IOUSBHostFamily::validateEndpointMaxPacketSize: USB 2.0 5.[5-8].3: endpoint 0x00 invalid wMaxPacketSize 0x0000

130345.651866 AppleUSB20HubPort@02241000: AppleUSBHostPort::enumerateDeviceComplete_block_invoke: enumeration failed

130345.651890 AppleUSB20HubPort@02241000: AppleUSBHostPort::terminateDevice: destroying 0x0000/0000/0000 (IOUSBHostDevice): enumeration failure

130345.651963 AppleUSB20HubPort@02241000: AppleUSBHostPort::disconnect: persistent enumeration failures

130345.651990 IOUSBHostDevice@02241000: IOUSBHostDevice::start_block_invoke: device will not be registered for matching

Does anyone have any ideas on the issue? I have tried CircuitPython 6 and the 7.1 beta. Nothing changed, though.

It seems to be an issue only when I have programmed the Pico as the Chord Player, which leads me to think it's an issue with either my code or one of the libraries (possibly between tinyUSB or Adafruit_CIRCUITPYTHON_MIDI...?). 

Sorry if this is the wrong place for this, thanks for any suggestions.

**Comment by hathach** [comment]
since you are using circuitpython, you should open an issue in https://github.com/adafruit/circuitpython . There is a lot more knowledgeable people with ciruicpython skill there. TinyUSB repo generally only resolve issue with bare minimal example

**Reply by johnnyw66** [reply]
FWIW, I have similar issues using MicroPython.
